cli/azd/pkg/tools/bicep/bicep.go

```diff
@@ -133,7 +117,12 @@ func (cli *bicepCli) Build(ctx context.Context, file string) (string, error) {
 }
 
 func (cli *bicepCli) runCommand(ctx context.Context, args ...string) (exec.RunResult, error) {
-	runArgs := exec.NewRunArgs("az", args...)
+	bicepPath, err := getBicepCliPath()
```

Comment:
Wondering..  is there a reason why we would prefer trying to find the bicep path on every `runCommand`?
Do we want to support something like, letting `azd` running and waiting for user to install bicep and then resume and try to run again?  (versus the current behavior of returning error and terminating azd) ?

If not, maybe we can move the `getBicepCliPath` to the `NewBicepCli` ?  and we keep the path within the cli state?

Reply:
I can not move it into NewBicepCli because we need to be able to create an instance of this type to call CheckInstalled on it (something Wei has mentioned is very weird and he'd like to fix later) but I can cache the result on the object so we don't compute it multiple times.